lib/types.nix

```diff
@@ -491,6 +492,14 @@ rec {
       merge = mergeEqualOption;
     };
 
+    pathInStore = mkOptionType {
+      name = "pathInStore";
+      description = "path in the Nix store";
+      descriptionClass = "noun";
+      check = x: isStringLike x && lib.hasPrefix builtins.storeDir (toString x);
```

Comment:
(I wish we had a proper subdirectory test in the path library.)

Reply:
Actually we probably only want to accept *strict* subdirectories here, so I'll just add a slash.